packages/fuels-programs/src/calls/traits/extendable.rs

```diff
@@ -0,0 +1,48 @@
+use fuels_core::types::bech32::Bech32ContractId;
+
+use crate::calls::{
+    utils::{new_variable_outputs, sealed},
+    ContractCall, ScriptCall,
+};
+
+pub trait Extendable: sealed::Sealed {
+    fn append_contract(&mut self, contract_id: Bech32ContractId);
+    fn append_variable_outputs(&mut self, num: u64);
```

Comment:
This one will be gone probably when the variable outputs pr lands. 

For the remaining two a good name might have something to do around the idea of being able to give the bindings of a different contract in order to call it and parse its logs.

Maybe: `ContractDependencyConfigurator` or `ContractDependencyIntegrator` or something like that.